### PR TITLE
reduce memory in ManifestMetadata

### PR DESCRIPTION
**Pull request by SimonCropp**
# Bug

Fixes: 

## Description

reduce memory in ManifestMetadata by using collection expressions that will result in those `IEnumerable` members being compiled to a `.Empty`.

let me know if u want to use explicit `Enumerable.Empty`s

## PR Checklist

- [ ] Meaningful title, helpful description and a linked NuGet/Home issue
- [ ] Added tests
- [ ] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs [hunk 1]
```diff
@@ -187,7 +187,7 @@ public Uri ProjectUrl
 
         public RepositoryMetadata Repository { get; set; }
 
-        private IEnumerable<PackageDependencyGroup> _dependencyGroups = new List<PackageDependencyGroup>();
+        private IEnumerable<PackageDependencyGroup> _dependencyGroups = [];
```
Comment (Nigusu-Allehu) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs:190]: nit: we could make the intent clear by
```suggestion
        private IEnumerable<PackageDependencyGroup> _dependencyGroups = Enumerable.Empty<PackageDependencyGroup>();
```

**Review by Nigusu-Allehu — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs:190]: Under the hood, they're the same thing in .NET Core.

**Review by nkolev92 — APPROVED** [review]
Thanks for the contribution @SimonCropp 

Something to consider. This change is not really in a hot path. ManifestMetadata is only done at pack time and given that pack runs in a separate process, the small gains here aren't winning us much. Small refactoring like this do have an impact on `git blame`, and in some cases, make things a bit more challenging to debug/understand intent behind certain changes, so when it comes cost/benefit, the hotter paths are better suited for these types of changes

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs (hunk 1 above)
Comment (SimonCropp) [reply src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs:190]: @Nigusu-Allehu there are a significant number of places in this code base where the intent is clear, but is results in sub-optimal code. the `[]` under the hood is smart enough to chose the best type for the majority of scenarios. IMO we should let it choose.

but it is your code base so happy to be overruled

**Comment by SimonCropp** [comment]
@nkolev92 

> Something to consider. This change is not really in a hot path. ManifestMetadata is only done at pack time and given that pack runs in a separate process, the small gains here aren't winning us much. Small refactoring like this do have an impact on git blame, and in some cases, make things a bit more challenging to debug/understand intent behind certain changes, so when it comes cost/benefit, the hotter paths are better suited for these types of changes

i guess there are 3 options

 * continue with small targeted PRS
 * do large PRs on the whole code base 
 * leave the slow code in place.

as for "This change is not really in a hot path". true. but it is more efficient and less code, given that it seems a shame to not apply the change no matter the code path

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs (hunk 1 above)
Comment (Nigusu-Allehu) [reply src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestMetadata.cs:190]: Either is fine with me as well. We can mark this as resolved

**Comment by nkolev92** [comment]
Hey @SimonCropp, 

I just wanted to again thank you for all the contributions lately, I see you've raised a bunch over the last weekend as well. 

With all the history and compatibility promises, we think there's benefit in preserving history to allow us to use `git blame` effectively.
We don't really have a blanket policy for refactoring changes, since it's all about the cost/benefit of each change, and with the amount of features in this repo and the different velocities of each, the comfort level for these changes will be different.

So a few things we'll consider:

- When changing coding patterns that improve performance for example, we prefer fixing and enabling a rule to never introduce the issue again. Seeing the amount of changes to enable the rule would allow to decide between more targeted/wholesale fixes.
- Non-functional refactoring PRs are not likely to be something we accept, especially in codepaths that are not very active, and haven't had changes recently. If there's a rule that can be enable to ensure the pattern is respected, we might considered it more strongly.

We want to be appreciative of your efforts, so before you spend a lot of effort on certain changes, please follow https://github.com/NuGet/NuGet.Client/blob/dev/CONTRIBUTING.md#contributing-step-by-step and file an issue to get a buy-off.

**Pull request merged**